[cutlass backend] Reduce log level for cutlass compilation error

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #153457
* __->__ #153397

Differential Revision: [D74596410](https://our.internmc.facebook.com/intern/diff/D74596410/)

This change should only affect cutlass backend. We realize that we are going to have Cuda compilation errors, and we do a really good job handling them and caching them. So reduce the logging levels there.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend] Reduce log level for cutlass compilation error



Differential Revision: [D74596410](https://our.internmc.facebook.com/intern/diff/D74596410/)

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/select_algorithm.py:
```diff
@@ -2130,9 +2131,20 @@ def wait_on_futures():
+                        futures[future], CUDATemplateCaller
+                    ):
+                        log.debug(
+                            "Exception %s for benchmark choice %s", e, futures[future]
```
```suggestion
                            "Exception %s for benchmark choice %s", e, futures[future], exc_info=True
```
Will ensure the same level of detail is printed as log.error

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass compilation error"


Differential Revision: [D74596410](https://our.internmc.facebook.com/intern/diff/D74596410/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass compilation error"


Differential Revision: [D74596410](https://our.internmc.facebook.com/intern/diff/D74596410/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/select_algorithm.py:
```diff
@@ -2238,10 +2253,13 @@ def benchmark_choices(
+                if not isinstance(choice, CUDATemplateCaller):
+                    log.error(
+                        "CUDA compilation error during autotuning: \n%s. \nIgnoring this choice.",
+                        str(e),
```
```suggestion
                        e,
```

henrylhtsang replied:
@Skylion007 I will make the change, but at this point it might be worth to change all str(e) in the codebase to e (at least for the logging cases)

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass compilation error"


Differential Revision: [D74596410](https://our.internmc.facebook.com/intern/diff/D74596410/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass compilation error"


Differential Revision: [D74596410](https://our.internmc.facebook.com/intern/diff/D74596410/)

This change should only affect cutlass backend. We realize that we are going to have Cuda compilation errors, and we do a really good job handling them and caching them. So reduce the logging levels there.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented on torch/_inductor/select_algorithm.py:
```diff
@@ -2238,10 +2253,13 @@ def benchmark_choices(
-                    "CUDA compilation error during autotuning: \n%s. \nIgnoring this choice.",
-                    str(e),
-                )
+                from torch._inductor.codegen.cuda.cuda_kernel import CUDATemplateCaller
```
I was told local imports like these can cause unwanted compile times.

henrylhtsang replied:
> I was told local imports like these can cause unwanted compile times.

I feel like if we try to import that at the top, it would cause circular import

henrylhtsang replied:
> give it a try and see? if not possible, it's probably fine since this is in exception handling path

actually I want to test it in a subsequent pR

henrylhtsang commented:
@pytorchbot merge
